de.bund.bfr.knime.fsklab.nodes/src/de/bund/bfr/knime/fsklab/v1_9/joiner/JoinerNodeModel.java

```diff
@@ -476,11 +477,8 @@ protected void loadJsonSetting() throws IOException, CanceledExecutionException,
    */
   private static List<Parameter> getModelParametersfromOldFile(File configFile)
       throws FileNotFoundException, IOException, ParseException {
-    JSONParser parser = new JSONParser();
-    Object obj = parser.parse(new FileReader(configFile));
-    JSONObject jsonObject = (JSONObject) obj;
-    JSONArray parameterList = (JSONArray) jsonObject.get("parameter");
-    return MAPPER.readValue(parameterList.toJSONString(), new TypeReference<List<Parameter>>() {});
+    JsonNode nodes = MAPPER.readTree(configFile);
+    return new ArrayList<>(Arrays.asList(MAPPER.treeToValue(nodes.get("parameter") , Parameter[].class)));
```

Comment:
It is not necessary to copy the list. We can return the list from Arrays.asList `Arrays.asList(MAPPER.treeToValue(nodes.get("parameter") , Parameter[].class))`

Reply:
the list should be mutable because I remove some elements from it in other places.
some files contain extra element which has to be removed before usage.